### PR TITLE
implement the edit roles

### PR DESCRIPTION
**Pull request by LinconOssa**

### CODE REVIEW

File: src/pages/privileges/outlets/roles/components/GeneralInformationForm/index.tsx [hunk 1]
```diff
@@ -66,39 +61,56 @@ export const GeneralInformationForm = forwardRef(
...
 
-    const handleSubmit = async () => {
-      setIsLoading(true);
-      const editedInfo = {
-        k_Rol: rol_id as string,
-        n_Rol: formik.values.roleName,
-        k_Tipcon: rol_id as string,
-        k_Aplica: formik.values.application,
-        n_Uso: formik.values.description,
+    // const handleSubmit = async () => {
```
Comment (carin91) [inline src/pages/privileges/outlets/roles/components/GeneralInformationForm/index.tsx:64]: delete every comment

File: src/pages/privileges/outlets/roles/components/GeneralInformationForm/interface.tsx [hunk 2]
```diff
@@ -146,7 +146,7 @@ export function GeneralInformationFormUI(props: GeneralInformationFormUIProps) {
               !hasChanges(formik.values) ||
               !(Object.keys(formik.errors).length === 0)
             }
-            handleSubmit={handleSubmit}
+            handleSubmit={() => {}}
```
Comment (carin91) [inline src/pages/privileges/outlets/roles/components/GeneralInformationForm/interface.tsx:149]: Refactor the general information form to delete its handle button and submit

File: src/pages/privileges/outlets/roles/components/InitializerForm/index.tsx [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+import { InitializerFormUI } from "./interface";
+import { IRol } from "../../types";
+import {
+  IAssignmentFormEntry,
+  IMessageState,
+} from "../../../users/types/forms.types";
+
+interface IInitializerForm {
+  dataOptionsForms: IAssignmentFormEntry[];
+  handleSubmit: (renderForm: IAssignmentFormEntry[]) => void;
```
Comment (carin91) [inline src/pages/privileges/outlets/roles/components/InitializerForm/index.tsx:13]: Delete the handle submit in the initializer form

**Review by carin91 — CHANGES_REQUESTED** [review]
The data loaded on the general information is not charging on the firs rendering.
image
Verify the state life cycle of this component

File: src/pages/privileges/outlets/roles/components/AncillaryAccountsForm/index.tsx [hunk 4]
```diff
@@ -101,7 +89,8 @@ export const AncillaryAccountsForm = forwardRef(function AncillaryAccountsForm(
   return (
     <AncillaryAccountsFormsUI
       formik={formik}
-      handleSubmit={handleSubmit}
+      handleChangeForm={handleChangeForm}
+      // handleSubmit={handleSubmit}
```
[the remark this answers is not in the document]
Comment (LinconOssa) [reply src/pages/privileges/outlets/roles/components/AncillaryAccountsForm/index.tsx]: Resolved

File: src/pages/privileges/outlets/roles/components/GeneralInformationForm/index.tsx (hunk 1 above)
Comment (LinconOssa) [reply src/pages/privileges/outlets/roles/components/GeneralInformationForm/index.tsx]: resolved

File: src/pages/privileges/outlets/roles/components/GeneralInformationForm/interface.tsx (hunk 2 above)
Comment (LinconOssa) [reply src/pages/privileges/outlets/roles/components/GeneralInformationForm/interface.tsx]: Resolved

File: src/pages/privileges/outlets/roles/components/AncillaryAccountsForm/index.tsx [hunk 5]
```diff
@@ -1,36 +1,42 @@
...
+    onHasChanges,
     initialValues,
-    rol_id,
+    handleSubmit,
     onSubmit,
     withSubmitButtons = false,
     handleAddRoleFormValid,
   } = props;
-  const [isLoading, setIsLoading] = useState(false);
+  const [isLoading] = useState(false);
```
[the remark this answers is not in the document]
Comment (LinconOssa) [reply src/pages/privileges/outlets/roles/components/AncillaryAccountsForm/index.tsx]: Resolved

File: src/pages/privileges/outlets/roles/components/InitializerForm/index.tsx (hunk 3 above)
Comment (LinconOssa) [reply src/pages/privileges/outlets/roles/components/InitializerForm/index.tsx]: -The handleSubmit is needed to provide a centralized and flexible point where all the necessary logic that needs to occur when the form is submitted or when its values ​​change can be handled. This includes additional validations, asynchronous calls, application state updates, and user notifications, among others.

I renamed it to

onFormValueChange:
• Indicates that the function is executed when there is a change in the form values.

-In short, onHasChanges is useful for maintaining communication between the form component and the component that contains it, allowing for greater flexibility and control over the state of the form and how changes are handled.

-these two are necessary

changeData

setChangedData

**Review by carin91 — APPROVED** [review]
